x/evm/statedb/statedb.go

```diff
@@ -71,6 +73,12 @@ func (s *StateDB) Keeper() Keeper {
 	return s.keeper
 }
 
+func (s *StateDB) Context() context.Context {
+	// s.ctx.CurrentCtx() replaced s.ctx after rebasing
+	// TODO(yevhenii): is it correct?
+	return s.ctx.CurrentCtx()
```

Comment:
Yup this is the correct ctx of the active snapshot to use for making state changes